airflow/jobs.py

```diff
@@ -2391,6 +2391,13 @@ def _execute_for_run_dates(self, run_dates, ti_status, executor, pickle_id,
         :param session: the current session object
         :type session: Session
         """
+        backfill_dagrun_order_reverse = conf.getboolean('scheduler',
+                                                        'backfill_dagrun_order_reverse')
+        ignore_depends_on_past = self.ignore_first_depends_on_past
+        is_reversed = backfill_dagrun_order_reverse and ignore_depends_on_past
+        if is_reversed:
+            # create dag run in reversed
+            run_dates = run_dates[::-1]
```

Comment:
Is there some flag on the dag that we look at to examine if the DAG does depend on past or not? Cos if not then not having to specify `-I` would be nice.

Seeing this behaviour/need I now think that this shouldn't be a config option.

Additionally if `-I` isn't passed then this config setting is silently ignored - that is going to be confusing, and this case should be an error.